chore: add 1.5.0 release notes (#2700)
defenseunicorns/uds-core: docs/operations/release-notes/1-5.mdx

---
title: UDS Core 1.5
description: UDS Core 1.5 release notes covering the unicorn flavor migration from RapidFort to Chainguard FIPS images and the new UDS Portal functional layer.
sidebar:
  order: 3.990
---

> [!NOTE]
> For general upgrade procedures, see the [Upgrade Overview](/operations/upgrades/overview/).

> UDS Core 1.5 migrates the unicorn flavor from RapidFort to Chainguard FIPS hardened images, introduces UDS Portal as a new optional functional layer in the upstream flavor, and stabilizes Keycloak on smaller nodes by disabling Infinispan virtual threads. Operators on the unicorn flavor must allow outbound access to `cgr.dev/defenseunicorns.com`, and RHEL 9 hosts running fapolicyd in enforcing mode need a rule to permit CNI execution from `/opt/cni/bin/.cgr/`.

Author: Good catch, fixed. Rewrote the closing sentence as conditional second-person guidance matching the style rule and the voice profile's inline "If X, then Y" pattern.

- **Keycloak: disable Infinispan virtual threads:** Infinispan 16 (shipped with Keycloak 26.6) enables virtual threads by default, which can deadlock under Kubernetes thread-pool starvation on nodes with fewer than 4 vCPUs. UDS Core now sets `-Dorg.infinispan.threads.virtual=false` to prevent this. References: [keycloak#48792](https://github.com/keycloak/keycloak/issues/48792), [keycloak#49203](https://github.com/keycloak/keycloak/issues/49203) ([#2686](https://github.com/defenseunicorns/uds-core/pull/2686)).

### Dependency updates

| Package                                | Previous | Updated                                                                                                              |
| -------------------------------------- | -------- | -------------------------------------------------------------------------------------------------------------------- |

> | Grafana                                | 12.4.2   | [13.0.1](https://github.com/grafana/grafana/releases/tag/v13.0.1)                                                    |

Author: Good catch, the major bump deserves a heads-up. Adding a Notable features bullet pointing at the v13 What's New page and calling out the operator-visible items (dynamic-dashboard auto-migration, image renderer removal, numeric `id` data source references disabled, removed `@grafana/ui` components). Keeping it out of the Breaking changes table since UDS Core itself ships clean and there's no required action for the default deployment, only for operators with custom dashboards or plugins.

| kube-prometheus-stack Helm chart       | 84.5.0   | [85.2.2](https://github.com/prometheus-community/helm-charts/releases/tag/kube-prometheus-stack-85.2.2)              |
| prometheus-operator-crds Helm chart    | 28.0.1   | [29.0.0](https://github.com/prometheus-community/helm-charts/releases/tag/prometheus-operator-crds-29.0.0)           |
| prometheus-blackbox-exporter Helm chart | 11.9.2   | [11.10.0](https://github.com/prometheus-community/helm-charts/releases/tag/prometheus-blackbox-exporter-11.10.0)     |
| Kube State Metrics                     | v2.18.0  | [v2.19.0](https://github.com/kubernetes/kube-state-metrics/releases/tag/v2.19.0)                                     |

## Upgrade considerations

The unicorn flavor change from RapidFort to Chainguard constitutes a breaking change but requires no steps beyond the standard upgrade procedure, except for fapolicyd users on RHEL 9 (see the Breaking changes table above).

## Related documentation

- [Upgrade Overview](/operations/upgrades/overview/) - general upgrade procedures and checklists

> - [UDS Core 1.5.0 Changelog](https://github.com/defenseunicorns/uds-core/blob/main/CHANGELOG.md#150-2026-05-27) - full changelog

Author: The CHANGELOG anchor is the established pattern in this repo — 0.60 through 1.3 all use the same `CHANGELOG.md#<version>-<date>` format. The link is correct as written; no change needed.

> - [UDS Core 1.5.0 Changelog](https://github.com/defenseunicorns/uds-core/blob/main/CHANGELOG.md#150-2026-05-27) - full changelog

Author: The CHANGELOG anchor is the established pattern in this repo — 0.60 through 1.3 all use the same `CHANGELOG.md#<version>-<date>` format. The link is correct as written; no change needed.

---
title: UDS Core 1.5
description: UDS Core 1.5 release notes covering the unicorn flavor migration from RapidFort to Chainguard FIPS images and the new UDS Portal functional layer.
sidebar:
  order: 3.990
---

> [!NOTE]
> For general upgrade procedures, see the [Upgrade Overview](/operations/upgrades/overview/).

> UDS Core 1.5 migrates the unicorn flavor from RapidFort to Chainguard FIPS hardened images, introduces UDS Portal as a new optional functional layer in the upstream flavor, and stabilizes Keycloak on smaller nodes by disabling Infinispan virtual threads. If you run the unicorn flavor, you must allow outbound access to `cgr.dev/defenseunicorns.com`. If your RHEL 9 hosts run fapolicyd in enforcing mode, you also need a rule to permit CNI execution from `/opt/cni/bin/.cgr/`.

Reviewer: ```suggestion
UDS Core 1.5 migrates the unicorn flavor from RapidFort to Chainguard FIPS hardened images, introduces UDS Portal as a new optional functional layer in the upstream/unicorn flavor, and stabilizes Keycloak on smaller nodes by disabling Infinispan virtual threads. If you run the unicorn flavor, you must allow outbound access to `cgr.dev/defenseunicorns.com`. If your RHEL 9 hosts run fapolicyd in enforcing mode, you also need a rule to permit CNI execution from `/opt/cni/bin/.cgr/`.
```

Reviewer: fixed.

> UDS Core 1.5 migrates the unicorn flavor from RapidFort to Chainguard FIPS hardened images, introduces UDS Portal as a new optional functional layer in the upstream flavor, and stabilizes Keycloak on smaller nodes by disabling Infinispan virtual threads. If you run the unicorn flavor, you must allow outbound access to `cgr.dev/defenseunicorns.com`. If your RHEL 9 hosts run fapolicyd in enforcing mode, you also need a rule to permit CNI execution from `/opt/cni/bin/.cgr/`.

Reviewer: > If you run the unicorn flavor, you must allow outbound access to `cgr.dev/defenseunicorns.com`.
People consuming core shouldn't need to do this. 

I would remove this line. Images will be bundled in the zarf package and does not require outbound connection

Reviewer: fixed

| ------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |

> | Unicorn flavor images move from RapidFort to Chainguard FIPS images | All unicorn images are now pulled from `cgr.dev/defenseunicorns.com/*-fips` instead of `quay.io/rfcurated/*`. Clusters on the unicorn flavor need outbound network access to `cgr.dev/defenseunicorns.com`.         | Allow outbound network access to `cgr.dev/defenseunicorns.com` from cluster nodes. On RHEL 9 with fapolicyd in enforcing mode, add a rule to allow execution from `/opt/cni/bin/.cgr/` because the Chainguard CNI image writes `libcrypto.so.3` into that path at runtime. The RKE2 user data script handles this automatically. |

Reviewer: I wouldn't consider the swap from RF to Chainguard as a breaking change.  Also the statement about outbound access on clusters is false. Zarf pushes the images and clusters don't need to outreach.

I would remove this breaking changes section altogether as these aren't really breaking changes.

Reviewer: fixed

### Notable features

> - **UDS Portal as a new functional layer:** adds `packages/portal/` and `src/portal/` deploying [uds-portal](https://github.com/defenseunicorns/uds-portal) v0.2.0 as an optional layer. Available in the upstream flavor only. Follows the same `DOMAIN`/`ADMIN_DOMAIN` variable pattern as other layers ([#2699](https://github.com/defenseunicorns/uds-core/pull/2699)).

Reviewer: nit: would suggest cleaning this up. the domain sentence is not really helpful/relevant IMO
```suggestion
- **UDS Portal as a new functional layer:** adds `packages/portal/` and `src/portal/` deploying [uds-portal](https://github.com/defenseunicorns/uds-portal) v0.2.0 as an optional layer. Available in the upstream/unicorn flavor only ([#2699](https://github.com/defenseunicorns/uds-core/pull/2699)).
```

Reviewer: fixed

> - **Unicorn flavor: Chainguard FIPS images:** the unicorn flavor migrates from RapidFort to Chainguard FIPS hardened images. Chainguard images are built on distroless bases and satisfy FIPS 140-2 requirements. The Pepr `validateIstioImage` policy is updated to recognize `cgr.dev/defenseunicorns.com/istio-proxy-fips` as the valid unicorn proxy image. The Keycloak unicorn image requires an explicit JCE security provider override, which UDS Core injects internally via a ConfigMap and `JAVA_OPTS_APPEND`, so no operator action is required ([#2650](https://github.com/defenseunicorns/uds-core/pull/2650)).

Reviewer: nit: don't need to explain internal details here IMO. nothing changes for the end user here
```suggestion
- **Unicorn flavor: Chainguard FIPS images:** the unicorn flavor migrates from RapidFort to Chainguard FIPS hardened images. Chainguard images are built on distroless bases and satisfy FIPS 140-2 requirements ([#2650](https://github.com/defenseunicorns/uds-core/pull/2650)).
```

Reviewer: fixed

| Loki                                   | 3.7.1    | [3.7.2](https://github.com/grafana/loki/releases/tag/v3.7.2)                                                         |
| Velero                                 | 1.18.0   | [1.18.1](https://github.com/vmware-tanzu/velero/releases/tag/v1.18.1)                                                |
| kube-prometheus-stack Helm chart       | 84.5.0   | [85.2.2](https://github.com/prometheus-community/helm-charts/releases/tag/kube-prometheus-stack-85.2.2)              |
| prometheus-operator-crds Helm chart    | 28.0.1   | [29.0.0](https://github.com/prometheus-community/helm-charts/releases/tag/prometheus-operator-crds-29.0.0)           |
| prometheus-blackbox-exporter Helm chart | 11.9.2   | [11.10.0](https://github.com/prometheus-community/helm-charts/releases/tag/prometheus-blackbox-exporter-11.10.0)     |
| Kube State Metrics                     | v2.18.0  | [v2.19.0](https://github.com/kubernetes/kube-state-metrics/releases/tag/v2.19.0)                                     |

## Upgrade considerations

> The unicorn flavor change from RapidFort to Chainguard constitutes a breaking change but requires no steps beyond the standard upgrade procedure, except for fapolicyd users on RHEL 9 (see the Breaking changes table above).

Reviewer: I would say this is not a breaking change. I would just include the fapolicyd note for rhel 9/rke2 here intead.

```suggestion
On RHEL 9 with fapolicyd in enforcing mode, add a rule to allow execution from `/opt/cni/bin/.cgr/` because the Chainguard Istio CNI image writes `libcrypto.so.3` into that path at runtime.
```

Also maybe link istio pre-reqs document here: https://docs.defenseunicorns.com/core/getting-started/production/prerequisites/#istio-requirements which has this called out.

Reviewer: fixed
